Jellyfin.Server.Implementations/MediaSegments/MediaSegmentsManager.cs

```diff
@@ -0,0 +1,214 @@
+        {
+            if (found != null)
+            {
+                found.StartTicks = segment.StartTicks;
+                found.EndTicks = segment.EndTicks;
+                found.Action = segment.Action;
+                found.Comment = segment.Comment;
+            }
+            else
+            {
+                dbContext.Segments.Add(segment);
+            }
+        }
+
+        /// <summary>
+        /// Validate a segment: itemId, start >= end and type.
+        /// </summary>
+        /// <param name="segment">The segment to validate.</param>
+        private void ValidateSegment(MediaSegment segment)
+        {
+            if (segment.ItemId.Equals(default))
+            {
+                throw new ArgumentException($"itemId is default: itemId={segment.ItemId} for segment with type '{segment.Type}.{segment.TypeIndex}'");
+            }
+
+            if (segment.StartTicks >= segment.EndTicks)
+            {
+                throw new ArgumentException($"start >= end: {segment.StartTicks}>={segment.EndTicks} for segment itemId '{segment.ItemId}' with type '{segment.Type}.{segment.TypeIndex}'");
+            }
+        }
+
+        /// <summary>
+        /// Delete all segments when itemid is deleted from library.
+        /// </summary>
+        /// <param name="sender">The sending entity.</param>
+        /// <param name="itemChangeEventArgs">The <see cref="ItemChangeEventArgs"/>.</param>
+        private async void LibraryManagerItemRemoved(object? sender, ItemChangeEventArgs itemChangeEventArgs)
+        {
+            await DeleteSegmentsAsync(itemChangeEventArgs.Item.Id).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        public async Task<List<MediaSegment>> DeleteSegmentsAsync(Guid itemId, int? streamIndex = null, int? typeIndex = null, MediaSegmentType? type = null)
+        {
+            var allSegments = new List<MediaSegment>();
+
+            if (itemId.Equals(default))
+            {
+                throw new ArgumentException($"itemId is not set. Please provide one.");
+            }
+
+            var dbContext = await _dbProvider.CreateDbContextAsync().ConfigureAwait(false);
+            await using (dbContext.ConfigureAwait(false))
+            {
+                IQueryable<MediaSegment> queryable = dbContext.Segments.Where(s => s.ItemId.Equals(itemId));
+
+                if (!streamIndex.Equals(null))
+                {
+                    queryable = queryable.Where(s => s.StreamIndex.Equals(streamIndex));
+                }
+
+                if (!type.Equals(null))
+                {
+                    queryable = queryable.Where(s => s.Type.Equals(type));
+                }
+
+                if (!typeIndex.Equals(null))
+                {
+                    queryable = queryable.Where(s => s.TypeIndex.Equals(typeIndex));
+                }
+
+                allSegments = queryable.AsNoTracking().ToList();
```

Comment:
I don't think `RemoveRange()` works if you call `AsNoTracking()` here
```suggestion
                allSegments = await queryable.ToListAsync().ConfigureAwait(false);
```

Reply:
I tested it and it worked